### PR TITLE
geos 3.10.6

### PR DESCRIPTION
**Pull request by lorepirri**
geos 3.10.6

**Destination channel:** defaults

### Links

- [PKG-6068]
- dev_url (conda-forge): https://github.com/libgeos/geos/tree/3.10.6
- conda-forge: https://github.com/conda-forge/geos-feedstock/blob/main/recipe/meta.yaml
- conda-forge at version 3.10.3: https://github.com/conda-forge/geos-feedstock/tree/881e2f45065e17852a2c760d47691aaa6a652101
- pypi: _not there_

### Explanation of changes:

- bump version
- remove not needed patches
- update build scripts
- folder cleanup
- ⚠️ update the `conda_build_config.yaml` with the last `geos 3.10.6` version, after merging and deploying `geos`: https://github.com/AnacondaRecipes/aggregate/pull/325

[PKG-6068]: https://anaconda.atlassian.net/browse/PKG-6068?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
 source:
   url: https://github.com/libgeos/libgeos/archive/{{ version }}.tar.gz
-  sha256: e9e20e83572645ac2af0af523b40a404627ce74b3ec99727754391cdf5b23645
-  patches:
-    - 0001-fix-xmltester.patch
-    - 0002-remove-bad-test.patch  # [linux and not x86]
-#   - 0003-removed-bad-issue.patch  # [linux and not x86]
-    - 0004-remove-failing-geos-test.patch  # [linux and not x86]
-    - 0005-remove-failing-linnearref-test.patch  # [linux and not x86]
+  sha256: 42c9e048b7103ecfce784e82aef2bcd1afec340b4483295cb3ff3817960789b1
```
Comment (cbouss) [inline recipe/meta.yaml:9]: Can you add this patch too? https://github.com/conda-forge/geos-feedstock/blob/main/recipe/patches/0001-unconditional-geos-pc.patch

File: recipe/meta.yaml [hunk 2]
```diff
@@ -23,30 +18,27 @@ build:
...
+    - make   # [not win]
+    - ninja  # [win]
  
 test:
   commands:
     - geos-config --version  # [not win]
 
 about:
-  home: http://trac.osgeo.org/geos/
+  home: https://trac.osgeo.org/geos/
```
Comment (cbouss) [inline recipe/meta.yaml:32]:
```suggestion
  home: https://libgeos.org/
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -23,30 +18,27 @@ build:
...
   license: LGPL-2.1
+  license_family: LGPL
   license_file: COPYING
   summary: Geometry Engine - Open Source
   description: |
     GEOS (Geometry Engine - Open Source) is a C++ port of the Java Topology
     Suite (JTS). As such, it aims to contain the complete functionality of
     JTS in C++.
-  doc_url: http://trac.osgeo.org/geos/
+  doc_url: https://trac.osgeo.org/geos/
```
Comment (cbouss) [inline recipe/meta.yaml:41]:
```suggestion
  doc_url: https://libgeos.org/
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -23,30 +18,27 @@ build:
...
   dev_url: https://github.com/libgeos/libgeos
```
Comment (cbouss) [inline recipe/meta.yaml:42]:
```suggestion
  dev_url: https://github.com/libgeos/geos
```

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml:9]: Done! Thanks for the addition!

File: recipe/meta.yaml [hunk 5]
```diff
@@ -23,31 +20,30 @@ build:
...
+    - m2-patch  # [win]
+
 test:
   commands:
     - geos-config --version  # [not win]
 
 about:
-  home: http://trac.osgeo.org/geos/
+  home: https://libgeos.org/
   license: LGPL-2.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:37]: LGPL-2.1-only or LGPL-2.1-or-later?

Comment (lorepirri) [reply recipe/meta.yaml:37]: it seems `or-later` even though CF says `only`... `geos` forgot to change the copyright info eheh https://github.com/libgeos/geos/blob/3.10.6/COPYING#L473-L479

Comment (lorepirri) [reply recipe/meta.yaml]: set to `or-later`

**Review by skupr-anaconda — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Comment by lorepirri** [comment]
I updated the `conda_build_config.yaml` with the last geos `3.10.6` version

**Pull request merged**